src/SimpleWallet/SimpleWallet.cpp

```diff
@@ -2173,51 +2433,181 @@ bool simple_wallet::transfer(const std::vector<std::string> &args) {
 	}
 #endif
 
+    std::vector<TransactionMessage> messages;
+    for (auto dst : cmd.dsts) {
+      for (auto msg : cmd.messages) {
+        messages.emplace_back(TransactionMessage { msg, dst.address});
+      }
+    }
+
+    uint64_t ttl = 0;
+    if (cmd.ttl != 0) {
+      ttl = static_cast<uint64_t>(time(nullptr)) +  cmd.ttl;
+    }
+
+    std::string sender = m_wallet->getAddress();
+
+	logger(INFO) << "Senderaddress: " << sender;
+
+
     CryptoNote::WalletHelper::SendCompleteResultObserver sent;
 
     std::string extraString;
     std::copy(cmd.extra.begin(), cmd.extra.end(), std::back_inserter(extraString));
 
     WalletHelper::IWalletRemoveObserverGuard removeGuard(*m_wallet, sent);
 
-    CryptoNote::TransactionId tx = m_wallet->sendTransaction(cmd.dsts, cmd.fee, extraString, cmd.fake_outs_count, 0);
+    CryptoNote::TransactionId tx = m_wallet->sendTransaction(cmd.dsts, cmd.fee, extraString, cmd.fake_outs_count, 0, messages, ttl, sender);
     if (tx == WALLET_LEGACY_INVALID_TRANSACTION_ID) {
-      fail_msg_writer() << "Can't send money";
+      failMsgWriter() << "Can't send money";
       return true;
     }
 
     std::error_code sendError = sent.wait(tx);
     removeGuard.removeObserver();
 
     if (sendError) {
-      fail_msg_writer() << sendError.message();
+      failMsgWriter() << sendError.message();
       return true;
     }
 
     CryptoNote::WalletLegacyTransaction txInfo;
     m_wallet->getTransaction(tx, txInfo);
-    success_msg_writer(true) << "Money successfully sent, transaction id: " << Common::podToHex(txInfo.hash) << ", key: " << Common::podToHex(txInfo.secretKey);
+    successMsgWriter(true) << "Money successfully sent, transaction id: " << Common::podToHex(txInfo.hash) << ", key: " << Common::podToHex(txInfo.secretKey);
 
     try {
       CryptoNote::WalletHelper::storeWallet(*m_wallet, m_wallet_file);
     } catch (const std::exception& e) {
-      fail_msg_writer() << e.what();
+      failMsgWriter() << e.what();
       return true;
     }
   } catch (const std::system_error& e) {
-    fail_msg_writer() << e.what();
+    failMsgWriter() << e.what();
   } catch (const std::exception& e) {
-    fail_msg_writer() << e.what();
+    failMsgWriter() << e.what();
   } catch (...) {
-    fail_msg_writer() << "unknown error";
+    failMsgWriter() << "unknown error";
+  }
+
+  return true;
+}
+//----------------------------------------------------------------------------------------------------
+bool simple_wallet::sendMsg(const std::vector<std::string>& args) {
```

Comment:
Inside this function each tab character must be replaced with 4 spaces. Please, never use tabs for indentation!